src/constraint/spec.ts

```diff
@@ -534,6 +534,26 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
     }
   },
   {
+    name: 'omitNonLinearScaleTypeWithStack',
+    description: 'Stacked plot should only use linear scale',
+    properties: [Property.CHANNEL, Property.MARK, Property.AGGREGATE, Property.AUTOCOUNT, Property.SCALE, Property.SCALE_TYPE, Property.TYPE],
+    allowEnumSpecForProperties: false,
+    strict: true,
+    satisfy: (specM: SpecQueryModel, schema: Schema, opt: QueryConfig) => {
+      const stack = specM.stack();
+      if (stack) {
+        for (let encQ of specM.getEncodings()) {
+          if ((!!encQ.aggregate || encQ.autoCount === true) && (encQ.type === Type.QUANTITATIVE)) {
```

Comment:
I think this is only for x and y? 

For example, if you group things by detail for the stack and map color to another aggregate field -- color can use non-linear scale.

(Add this as a test case)